akka-stream/src/main/scala/akka/stream/impl/FanIn.scala

```diff
@@ -186,7 +186,7 @@ private[akka] object FanIn {
     }
 
     def inputsAvailableFor(id: Int) = new TransferState {
-      override def isCompleted: Boolean = depleted(id)
+      override def isCompleted: Boolean = depleted(id) || cancelled(id)
```

Comment:
No, when _we_ cancelled this input, this input cannot possibly contribute to our success anymore, hence `isCompleted`.